staker/validator_wallet.go

```diff
@@ -283,20 +302,78 @@ func (v *ContractValidatorWallet) ExecuteTransactions(ctx context.Context, build
 	if err != nil {
 		return nil, err
 	}
-	arbTx, err := v.con.ExecuteTransactionsWithGasRefunder(auth, gasRefunder, data, dest, amount)
+	txData, err := validatorABI.Pack("executeTransactionsWithGasRefunder", gasRefunder, data, dest, amount)
+	if err != nil {
+		return nil, fmt.Errorf("packing arguments for executeTransactionWithGasRefunder: %w", err)
+	}
+	gas, err := v.gasForTxData(ctx, auth, txData)
+	if err != nil {
+		return nil, fmt.Errorf("getting gas for tx data: %w", err)
+	}
+	arbTx, err := v.dataPoster.PostTransaction(ctx, time.Now(), auth.Nonce.Uint64(), nil, *v.Address(), txData, gas, auth.Value)
 	if err != nil {
 		return nil, err
 	}
 	builder.transactions = nil
 	return arbTx, nil
 }
 
+func (v *ContractValidatorWallet) estimateGas(ctx context.Context, value *big.Int, data []byte) (uint64, error) {
+	h, err := v.l1Reader.LastHeader(ctx)
+	if err != nil {
+		return 0, fmt.Errorf("getting the last header: %w", err)
+	}
+	gasFeeCap := new(big.Int).Mul(h.BaseFee, big.NewInt(2))
+	gasFeeCap = arbmath.BigMax(gasFeeCap, arbmath.FloatToBig(params.GWei))
+
+	gasTipCap, err := v.l1Reader.Client().SuggestGasTipCap(ctx)
+	if err != nil {
+		return 0, fmt.Errorf("getting suggested gas tip cap: %w", err)
+	}
+	gp, err := v.l1Reader.Client().SuggestGasPrice(ctx)
+	if err != nil {
+		return 0, fmt.Errorf("getting suggested gas price: %w", err)
+	}
+	g, err := v.l1Reader.Client().EstimateGas(
+		ctx,
+		ethereum.CallMsg{
+			From:      v.auth.From,
+			To:        v.Address(),
+			Value:     value,
+			Data:      data,
+			GasFeeCap: gasFeeCap,
+			GasTipCap: gasTipCap,
+			GasPrice:  gp,
```

Comment:
Specifying GasPrice is redundant if you already specify GasFeeCap and GasTipCap

Reply:
Done.